tensorflow/contrib/layers/python/layers/layers.py

```diff
@@ -836,6 +837,92 @@ def fully_connected(inputs,
 
 
 @add_arg_scope
+def layer_norm(inputs,
+               center=True,
+               scale=False,
+               epsilon=0.001,
+               activation_fn=None,
+               reuse=None,
+               variables_collections=None,
+               outputs_collections=None,
+               trainable=True,
+               scope=None):
+  """Adds a Layer Normalization layer from https://arxiv.org/abs/1607.06450.
+
+    "Layer Normalization"
+
+    Jimmy Lei Ba, Jamie Ryan Kiros, Geoffrey E. Hinton
+
+  Can be used as a normalizer function for conv2d and fully_connected.
+
+  Args:
+    inputs: a tensor of size `[batch_size, height, width, channels]`
+            or `[batch_size, channels]`.
+    center: If True, subtract `beta`. If False, `beta` is ignored.
+    scale: If True, multiply by `gamma`. If False, `gamma` is
+      not used. When the next layer is linear (also e.g. `nn.relu`), this can be
+      disabled since the scaling can be done by the next layer.
+    epsilon: small float added to variance to avoid dividing by zero.
```

Comment:
Is the variance is zero, you don't have to change the values since they're all the same. Can you do that instead: if variance < float_epsilon, do not divide by it? You can find the appropriate epsilon by asking python, and then remove this argument.